chat/src/main/java/com/github/twitch4j/chat/TwitchChatConnectionPool.java

```diff
@@ -0,0 +1,163 @@
+    @Builder.Default
+    protected final Supplier<OAuth2Credential> chatAccount = () -> null;
+
+    /**
+     * Sends the specified message to the channel, if it has been subscribed to.
+     *
+     * @param channel the channel to send the message to
+     * @param message the message to send
+     * @return whether a {@link TwitchChat} instance subscribed to that channel was identified and used
+     */
+    public boolean sendMessage(final String channel, final String message) {
+        return this.sendMessage(channel, channel, message);
+    }
+
+    /**
+     * Sends a message from the {@link TwitchChat} identified either to a channel or directly on the socket.
+     *
+     * @param channelToIdentifyChatInstance the channel used to identify which {@link TwitchChat} instance should be used to send the message; the instance must be subscribed to this channel.
+     * @param targetChannel                 the channel to send the message to, if not null (otherwise it is sent directly on the socket)
+     * @param message                       the message to be sent
+     * @return whether a {@link TwitchChat} instance was found and used to send the message
+     */
+    public boolean sendMessage(final String channelToIdentifyChatInstance, final String targetChannel, final String message) {
+        if (channelToIdentifyChatInstance == null)
+            return false;
+
+        final TwitchChat chat = subscriptions.get(channelToIdentifyChatInstance.toLowerCase());
+        if (chat == null)
+            return false;
+
+        if (targetChannel != null)
+            chat.sendMessage(targetChannel, message);
+        else
+            chat.sendRaw(message);
+
+        return true;
+    }
+
+    /**
+     * Sends a whisper.
+     *
+     * @param channelToIdentifyChatInstance the channel used to identify which {@link TwitchChat} instance should be used to send the message; the instance must be subscribed to this channel.
+     * @param toChannel                     the channel to send the whisper to
+     * @param message                       the message to send in the whisper
+     * @return whether a {@link TwitchChat} instance was identified to send the message from
+     * @throws NullPointerException if the identified {@link TwitchChat} does not have a valid chatCredential
+     */
+    public boolean sendPrivateMessage(final String channelToIdentifyChatInstance, final String toChannel, final String message) {
+        final TwitchChat chat;
+        if (channelToIdentifyChatInstance == null || (chat = subscriptions.get(channelToIdentifyChatInstance.toLowerCase())) == null)
+            return false;
+
+        chat.sendPrivateMessage(toChannel, message);
+        return true;
+    }
+
+    /**
+     * Joins a channel.
+     *
+     * @param s the channel name
+     * @return the channel name
+     */
+    @Override
+    public String subscribe(String s) {
+        return super.subscribe(s != null ? s.toLowerCase() : null);
```

Comment:
How should failures be handled (e.g. channel does not exist or user is banned from the chat)?

Reply:
We could immediately issue a leave or provide a boolean option to do nothing (could be useful as if the user is unbanned, for example, upon the next organic websocket reconnection, messages would be received again)